src/security/cli_security.c

```diff
@@ -79,28 +87,38 @@ request_credentials_via_drpc(Drpc__Response **response)
 	if (request == NULL) {
 		D_ERROR("Couldn't allocate dRPC call\n");
 		drpc_close(agent_socket);
+		free(sock_path);
 		return -DER_NOMEM;
 	}
 
 	rc = drpc_call(agent_socket, R_SYNC, request, response);
 
 	drpc_close(agent_socket);
 	drpc_call_free(request);
+	free(sock_path);
 	return rc;
 }
 
 static char *
 get_agent_socket_path(void)
 {
+	char *path = NULL;
 	/*
 	 * UDS path may be set in an environment variable.
 	 */
-	char *path = getenv(DAOS_AGENT_DRPC_SOCK_ENV);
+	char *envpath = getenv(DAOS_AGENT_DRPC_DIR_ENV);
 
-	if (path == NULL) {
+	if (envpath == NULL) {
 		path = DEFAULT_DAOS_AGENT_DRPC_SOCK;
 	}
 
+	/*
+	 * Craft our socket path from the directory.
+	 */
+	if (asprintf(&path, "%s/agent.sock", envpath) < 0) {
```

Comment:
Do we want to define the socket name "agent.sock"in the header, too? Could share it between C and Go code